### PR TITLE
Vue wallet services plugin composable

### PR DESCRIPTION
**Pull request by BboyStatix**
## Motivation and Context

Jira Link: https://toruslabs.atlassian.net/browse/PD-3899

## Description

Adds wallet services plugin to vue composables.

## How has this been tested?

Tried it out via demo app to see if the methods are working

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project. (run lint)
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/composables/modal-vue-composables/src/Web3AuthProvider.ts [hunk 1]
```diff
@@ -215,7 +213,7 @@ export const Web3AuthProvider = defineComponent({
       { immediate: true }
     );
 
-    provide(Web3AuthContextKey, {
+    provide<IWeb3AuthContext>("web3auth_context", {
```
Comment (chaitanyapotti) [inline packages/composables/modal-vue-composables/src/Web3AuthProvider.ts:216]: make this a constant

File: demo/vue-app-modal-vue-composables/.gitignore [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Logs
```
Comment (chaitanyapotti) [inline demo/vue-app-modal-vue-composables/.gitignore:1]: remove this example pls. it's hard to maintain so many

File: packages/hooks/wallet-services-plugin-react-hooks/src/context/WalletServicesContext.ts [hunk 3]
```diff
@@ -11,6 +11,14 @@ export function WalletServicesContextProvider<T extends IBaseWeb3AuthHookContext
   const [isPluginConnected, setIsPluginConnected] = useState<boolean>(false);
   const [walletServicesPlugin, setWalletServicesPlugin] = useState<WalletServicesPlugin>(null);
   const web3AuthContext = useContext(context);
+  const { getPlugin, isInitialized } = web3AuthContext;
+
+  useEffect(() => {
```
Comment (chaitanyapotti) [inline packages/hooks/wallet-services-plugin-react-hooks/src/context/WalletServicesContext.ts:16]: should also add a check for isConnected

Comment (BboyStatix) [reply packages/hooks/wallet-services-plugin-react-hooks/src/context/WalletServicesContext.ts:16]: Screenshot 2024-09-17 at 6 48 08 PM
When web3auth is not connected, the plugin event is not fired i.e.
```javascript
const connectedListener = () => {
  isPluginConnected.value = true;
};

newWalletServicesPlugin.on(PLUGIN_EVENTS.CONNECTED, connectedListener);
```
Therefore, it is adequate to add plugin and set up listeners on intiialised.
Unless you are meaning that I should add it even prior to initialisation.

Comment (BboyStatix) [reply packages/hooks/wallet-services-plugin-react-hooks/src/context/WalletServicesContext.ts:16]: But at the same time plugin could be added via `addPlugin` method even after connected. So how to listen for that?

**Review by chaitanyapotti — APPROVED** [review]
lgtm

**Pull request merged**